docs/source/notes/modules.rst

```diff
@@ -0,0 +1,433 @@
+:mod:`torch.nn` namespace for computation that is commonly found within neural networks, including pooling,
+convolutions, loss functions, etc.
+
+In the next section, we give a full example of training a neural network.
+
+For more information, check out:
+
+* Recursively :func:`~torch.nn.Module.apply` a function to a module and its submodules
+* Library of PyTorch-provided modules: `torch.nn <https://pytorch.org/docs/stable/nn.html>`_
+* Defining neural net modules: https://pytorch.org/tutorials/beginner/examples_nn/two_layer_net_module.html
+
+Neural Network Training with Modules
+------------------------------------
+
+Once a network is built, it has to be trained, and its parameters can be easily optimized with one of PyTorch’s
+optimizers from :mod:`torch.optim`:
+
+.. code-block:: python
+
+   # Create the network (from previous section) and optimizer
+   net = Net()
+   optimizer = torch.optim.SGD(net.parameters(), lr=1e-4, weight_decay=1e-2, momentum=0.9)
+
+   # Run a sample training loop that "teaches" the network
+   # to output the constant zero function
+   for _ in range(10000):
+     input = torch.randn(4)
+     output = net(input)
+     loss = torch.abs(output)
+     net.zero_grad()
+     loss.backward()
+     optimizer.step()
+
+In this simplified example, the network simply learns to output zero, which is not very interesting, but the
+key parts of training are present:
+
+* A network is created.
+* An optimizer (in this case, a stochastic gradient descent optimizer) is created, and the network’s
+  parameters are associated with it.
+* A training loop...
+    * acquires an input,
+    * runs the network,
+    * computes a loss,
+    * zeros the network’s parameters’ gradients,
+    * calls loss.backward() to update the parameters’ gradients,
+    * calls optimizer.step() to apply the gradients to the parameters.
+
+After the above snippet has been run, note that the network's parameters have changed. In particular, examining the
+value of ``l1``\ 's ``weight`` parameter shows that its values are now much closer to 0 (as may be expected):
+
+.. code-block:: python
+
+   print(net.l1.weight)
+   : Parameter containing:
+   tensor([[-0.0013],
+           [ 0.0030],
+           [-0.0008]], requires_grad=True)
+
+Training neural networks can often be tricky. For more information, check out:
+
+* Using optimizers: https://pytorch.org/tutorials/beginner/examples_nn/two_layer_net_optim.html.
+* Neural network training: https://pytorch.org/tutorials/beginner/blitz/neural_networks_tutorial.html
+* Introduction to autograd: https://pytorch.org/tutorials/beginner/blitz/autograd_tutorial.html
+
+Module State
+------------
+
+In the previous section, we demonstrated training a module's "parameters", or learnable aspects of computation.
+Now, if we want to save out the trained model, we can do so by saving its ``state_dict`` (i.e. "state dictionary"):
```

Comment:
save out?  save the trained model to disk?

Reply:
Changed: "...if we want to save out the trained model..." -> "...if we want to save the trained model to disk..."